src/MatrixTypes_Native.f90

```diff
@@ -691,11 +692,10 @@ SUBROUTINE assemble_BandedMatrixType(thisMatrix)
   INTEGER(SLK),ALLOCATABLE :: diagRank(:)
 
   REQUIRE(thisMatrix%isInit)
-  REQUIRE(thisMatrix%nnz == thisMatrix%counter)
 
-  ALLOCATE(diagRank(SIZE(thisMatrix%iTmp)))
-  ALLOCATE(idxOrig(SIZE(thisMatrix%iTmp)))
-  DO i=1,SIZE(thisMatrix%iTmp)
+  ALLOCATE(diagRank(thisMatrix%counter))
+  ALLOCATE(idxOrig(thisMatrix%counter))
+  DO i=1,thisMatrix%counter
```

Comment:
What's the significance of changing from `thisMatrix%iTmp` to `thisMatrix%counter`?

Reply:
Essentially this means it will loop over the added entries instead of the full iTmp array.
iTmp is an allocated array which stores up to nnz entries, but now because of these changes it may not be full.
So we loop up to counter, which counts the number of entries added prior to the assemble call (which makes the matrix structure read-only)

Reply:
Got it, so `%counter` is the number of entries set on the `%iTmp` array.